### PR TITLE
EN-238 Add WebSpotlight support via Management Api

### PR DESCRIPTION
**Pull request by arguit**
### Motivation

As a part of MAPI Improvements this will add support for activation and deactivation of Web Spotlight.

### Checklist

- [ ] Code follows coding conventions held in this repo
- [ ] Automated tests have been added
- [ ] Tests are passing
- [ ] Docs have been updated (if applicable)
- [ ] Temporary settings (e.g. variables used during development and testing) have been reverted to defaults

### How to test

If manual testing is required, what are the steps?

### CODE REVIEW

File: Kontent.Ai.Management.Tests/ManagementClientTests/WebSpotlightTests.cs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+namespace Kontent.Ai.Management.Tests.ManagementClientTests;
+
+public class WebSpotlightTests : IClassFixture<FileSystemFixture>
+{
+    private static readonly string WebSpotlightBaseUrl = $"{Endpoint}/projects/{ENVIRONMENT_ID}/web-spotlight";
+    private readonly Scenario _scenario = new(folder: "WebSpotlight");
+
+    [Fact]
+    public async void ActivateWebSpotlight_Returns_EnabledStatusAndRootTypeId()
```
Comment (pokornyd) [inline Kontent.Ai.Management.Tests/ManagementClientTests/WebSpotlightTests.cs:17]: nitpick: I checked other tests and we are not 100% consistent on signatures for xunit tests. some use `async Task`, others `async void`. if I understand, xunit has custom context which allows test method signatures with `async void`, so this is not really a problem, rather my observation :)

File: Kontent.Ai.Management/IManagementClient.cs [hunk 2]
```diff
@@ -803,4 +804,23 @@ public interface IManagementClient
     /// <param name="previewConfiguration">Represents configuration that will be used for project.</param>
     /// <returns>The <see cref="PreviewConfigurationModel"/> instance that represents the preview configuration.</returns>
     Task<PreviewConfigurationModel> ModifyPreviewConfigurationAsync(PreviewConfigurationModel previewConfiguration);
+
+    /// <summary>
+    /// Activates the web spotlight status with possibility to provide existing Root Type ID.
```
Comment (pokornyd) [inline Kontent.Ai.Management/IManagementClient.cs:809]: I would adjust the wording here a bit:

`Activates the web spotlight, allowing you to specify an existing Root Type ID.`

what do you think?

File: Kontent.Ai.Management/IManagementClient.cs [hunk 3]
```diff
@@ -803,4 +804,23 @@ public interface IManagementClient
...
+    /// <summary>
+    /// Deactivates the web spotlight.
+    /// </summary>
+    /// <returns>The <see cref="WebSpotlightModel"/> instance that represents the web spotlight status.</returns>
+    Task<WebSpotlightModel> DeactivateWebSpotlightAsync();
+
+    /// <summary>
+    /// Returns the web spotlight status.
+    /// </summary>
+    /// <returns>The <see cref="WebSpotlightModel"/> instance that represents the web spotlight status.</returns>
```
Comment (pokornyd) [inline Kontent.Ai.Management/IManagementClient.cs:824]: ditto in previous comment

**Review by pokornyd — DISMISSED** [review]

File: Kontent.Ai.Management/IManagementClient.cs (hunk 3 above)
Comment (arguit) [reply Kontent.Ai.Management/IManagementClient.cs]: Done.

File: Kontent.Ai.Management/IManagementClient.cs (hunk 2 above)
Comment (arguit) [reply Kontent.Ai.Management/IManagementClient.cs]: Done.

**Review by pokornyd — APPROVED** [review]

**Pull request merged**